### PR TITLE
Parse lbr records

### PR DESCRIPTION
**Pull request by johnikeson**

### CODE REVIEW

File: p4dlog.go [hunk 1]
```diff
@@ -76,6 +76,11 @@ func toInt64(buf string) (n int64) {
 	return
 }
 
+type lbrRegex struct {
```
Comment (rcowham) [inline p4dlog.go:79]: Don't think this belons in main module - only test module.

File: p4dlog.go [hunk 2]
```diff
@@ -115,47 +120,71 @@ func (block *Block) addLine(line string, lineNo int64) {
...
+	RPCHimarkFwd            int64     `json:"rpcHimarkFwd"`
+	RPCHimarkRev            int64     `json:"rpcHimarkRev"`
+	RPCSnd                  float32   `json:"rpcSnd"`
+	RPCRcv                  float32   `json:"rpcRcv"`
+	NetFilesAdded           int64     `json:"netFilesAdded"` // Valid for syncs and network estimates records
+	NetFilesUpdated         int64     `json:"netFilesUpdated"`
+	NetFilesDeleted         int64     `json:"netFilesDeleted"`
+	NetBytesAdded           int64     `json:"netBytesAdded"`
+	NetBytesUpdated         int64     `json:"netBytesUpdated"`
+	LbrRcsOpens             int64     `json:"lbrRcsOpens"`
```
Comment (rcowham) [inline p4dlog.go:158]: Add a comment, e.g. // Valid for sync/transmit commands

File: p4dlog.go [hunk 3]
```diff
@@ -528,6 +696,78 @@ func (c *Command) updateFrom(other *Command) {
 			c.Tables[k] = t
 		}
 	}
+	if c.LbrRcsOpens == 0 {
```
Comment (rcowham) [inline p4dlog.go:699]: Please copy convention above:

if other.SomeValue > 0 {
   c.SomeValue = other.SomeValue
}

What you have will not update.

File: p4dlog.go [hunk 4]
```diff
@@ -735,9 +975,16 @@ var trackChange = "--- change"
...
 var trackStorage = "--- storageup/"
+var trackLbrRcs = "--- lbr Rcs"
+var trackLbrCompress = "--- lbr Compress"
+var trackLbrUncompress = "--- lbr Uncompress"
 var reCmdTrigger = regexp.MustCompile(` trigger ([^ ]+)$`)
 var reTriggerLapse = regexp.MustCompile(`^lapse (\d+\.\d+)s|^lapse (\.\d+)s|^lapse (\d+)s`)
 var prefixTrackRPC = "--- rpc msgs/size in+out "
+var prefixTrackLbr = "---   opens+closes"
+var prefixTrackLbr2 = "---   reads+readbytes"
+var reTrackLbr = regexp.MustCompile(`^---\s*opens\+closes\+checkins\+exists\s+(\d+)\s*\+\s*(\d+)\s*\+\s*(\d+)\s*\+\s*(\d+)`)
```
Comment (rcowham) [inline p4dlog.go:986]: ^---\s*opens

Should be explicit about number of spaces, e.g 3 spaces.

^---   opens

Comment (rcowham) [reply p4dlog.go:986]: Also the other occurrences of \s* should be replaced.

File: p4dlog.go [hunk 5]
```diff
@@ -851,6 +1099,86 @@ func (fp *P4dFileParser) processTrackRecords(cmd *Command, lines []string) {
...
+					continue
+
+				}
+
+			}
+			m = reTrackLbrReadWrite.FindStringSubmatch(line)
+			if len(m) > 0 {
+				if strings.HasPrefix(line, prefixTrackLbr2) {
+					cmd.setLbrCompressReadWrites(m[1], m[3], processReadWriteBytes(m[2]), processReadWriteBytes(m[4]))
+
```
Comment (rcowham) [inline p4dlog.go:1154]: Remove unncessary blank lines please.

File: p4dlog.go [hunk 6]
```diff
@@ -930,6 +1258,26 @@ func (fp *P4dFileParser) processTrackRecords(cmd *Command, lines []string) {
 	fp.addCommand(cmd, hasTrackInfo)
 }
 
+func processReadWriteBytes(readByteValue string) int64 {
```
Comment (rcowham) [inline p4dlog.go:1261]: Rename "readByteValue" to "value" for clarity

File: p4dlog.go [hunk 7]
```diff
@@ -930,6 +1258,26 @@ func (fp *P4dFileParser) processTrackRecords(cmd *Command, lines []string) {
...
+	} else if l == "M" {
+		rtnVal = int64(float32(s * 1024 * 1024))
+	} else if l == "G" {
+		rtnVal = int64(float32(s * 1024 * 1024 * 1024))
+	} else if l == "T" {
+		rtnVal = int64(float32(s * 1024 * 1024 * 1024 * 1024))
+	} else if l == "P" {
+		rtnVal = int64(float32(s * 1024 * 1024 * 1024 * 1024 * 1024))
+	} else {
+		rtnVal = int64(float32(s))
```
Comment (rcowham) [inline p4dlog.go:1276]: In this case you are missing the last char in the string. Try parsing the whole string in this instance.

File: p4dlog_test.go
Comment (rcowham) [inline p4dlog_test.go:1]: Please add a new test which copies an existing test with records for db.someTable as well as all 3 of the new sets of info - make sure the logica copes with everything set.

Vary the values of a+b+c+d to make sure they are suitable copied.

--- lbr Rcs
---   opens+closes+checkins+exists 1+2+3+4
---   reads+readbytes+writes+writebytes 1+12.1K+0+3.3K
--- lbr Compress
---   opens+closes+checkins+exists 1+0+0+4
---   reads+readbytes+writes+writebytes 6+12.1K+0+3.3K
--- lbr Uncompress
---   opens+closes+checkins+exists 1+0+0+4
---   reads+readbytes+writes+writebytes 6+12.1K+0+3.3K

**Review by rcowham — CHANGES_REQUESTED** [review]

File: p4dlog.go (hunk 2 above)
Comment (johnikeson) [reply p4dlog.go:158]: something like this // Required for processing lbr records?

**Pull request closed without merging**